src/ssl/support.cc

```diff
@@ -1047,67 +1174,45 @@ issuerExistInCaDb(X509 *cert, const Security::ContextPointer &connContext)
     return gotIssuer;
 }
 
-const char *
-Ssl::uriOfIssuerIfMissing(X509 *cert, Security::CertList const &serverCertificates, const Security::ContextPointer &context)
+bool
+Ssl::issuerIsMissing(X509 *cert, const STACK_OF(X509) *serverCertificates, const Security::ContextPointer &context)
```

Comment:
If it is easy, let's return the pointer to the issuer certificate (or some other issuer information we can point to in all the "found" case) instead of bool, converting this function from "negative" into a positive one. We can call this public function findIssuerCertificate() or something like that.

Reply:
commit 13110313b6

> If it is easy

It was not so easy in the sense that I think a lot to return or not Security::CertPointer or raw X509 objects. The Security::CertPointer adds some unneeded certificates locking  if you only looking for the existence of the issuer.
The latest openSSL uses atomic operations to lock objects so I hope there is not an overhead.

The Security::CertPointer is selected, to be sure (force the programmers to check) that the returned object is always locked. Reason is the X509_STORE_CTX_get1_issuer which return a locked object (which must be locked to avoid release after this function completes). 

Changing the issuerIsMissing to be findIssuerCertificate force me to change also the issuerExistInCaDb and completeIssuers static functions.

I hope the result is enough good.